tools/packaging/common/envoy_bootstrap_v2.json

```diff
@@ -262,7 +262,11 @@
             ],
             "validation_context": {
               "trusted_ca": {
+                {{ if .sign_cert_at_kubernetes_ca }}
                 "filename": "./var/run/secrets/kubernetes.io/serviceaccount/ca.crt"
+                {{ else }}
+                "filename": "./etc/istio/citadel-ca-cert/ca-cert-ns.pem"
```

Comment:
what happens when a proxy tries to connect to a pilot that is configured differently? I am worried this is impossible to upgrade ?

Reply:
The TLS configurations for proxy and pilot must be consistent.

Reply:
how do you upgrade then? it sounds like it is impossible and we are back to the old "file mount to SDS" upgrade issue

Reply:
When upgrade, the config for envoy and pilot will be upgraded to have consistent TLS configuration. The methodology in this PR is no different from existing implementation in this file that uses kubernetes.io/service account/ca.crt.

Reply:
Its impossible to atomically update envoy's and Pilots. You will always have new pilot, then update the old envoys, so pilot needs to be backwards compatible with older envoy versions